### ISSUE TITLE
tmux: shell stderr: integer expression expected

### ISSUE THREAD

**Issue by vbauerster** [mawww/kakoune#1936]
I've noticed following error in \*debug\* buffer, appearing when inside tmux only.
Is it normal?

```
*** This is the debug buffer, where debug info will be written ***
shell stderr: <<<
/bin/sh: line 6: [: : integer expression expected
>>>
```
I use mac version, zsh as my default shell.

**Comment by lenormf** [comment]
It's not normal, please paste the contents of the buffer that is displayed upon running the following command:

```
$ kak -E 'set global debug shell' -e 'b *debug*'
```

**Comment by vbauerster** [comment]
Here you are:
```
*** This is the debug buffer, where debug info will be written ***
shell:

        if [ -n "$TMUX" ]; then
            VERSION_TMUX=$(tmux -V)
            VERSION_TMUX=$(expr "${VERSION_TMUX}" : 'tmux \([0-9]*\|master\)')

            if [ "${VERSION_TMUX}" = "master" ] \
                || [ "${VERSION_TMUX}" -ge 2 ]; then
                echo "
                    alias global repl tmux-repl-horizontal
                    alias global send-text tmux-send-text
                "
            else
                echo "
                    alias global repl tmux-repl-disabled
                    alias global send-text tmux-repl-disabled
                "
            fi
        fi
    
----
shell stderr: <<<
/bin/sh: line 6: [: : integer expression expected
>>>
shell:

        if [ "$TERM_PROGRAM" = "iTerm.app" -a -z "$TMUX" ]; then
            echo "
                alias global new iterm-new-vertical
                alias global focus iterm-focus
            "
        fi
    
----
shell:

        if [ -n "$TMUX" ]; then
            echo "
                alias global focus tmux-focus
                alias global new tmux-new-horizontal
            "
        fi
    
----
shell:

        [ -z "${STY}" ] && exit
        echo "
            alias global focus screen-focus
            alias global new screen-new-vertical
        "
    
----
shell:

        buffer_basename="${kak_buffile##*/}"
        buffer_dirname=$(dirname "${kak_buffile}")

        if [ -f "${kak_buffile}" ]; then
            newer=$(find "${buffer_dirname}"/".${buffer_basename}.kak."* -newer "${kak_buffile}" -exec ls -1t {} + 2>/dev/null | head -n 1)

            older=$(find "${buffer_dirname}"/".${buffer_basename}.kak."* \! -newer "${kak_buffile}" -exec ls -1t {} + 2>/dev/null | head -n 1)
        else
            # New buffers that were never written to disk.
            newer=$(ls -1t "${buffer_dirname}"/".${buffer_basename}.kak."* 2>/dev/null | head -n 1)
        fi

        if [ -z "${newer}" ]; then
            if [ -n "${older}" ]; then
                printf %s\\n "
                    echo -debug Old backup file(s) found: will not restore ${older} .
                "
            fi
            exit   
        fi

        printf %s\\n "
            ## Replace the content of the buffer with the content of the backup file
            echo -debug Restoring file: ${newer}

            execute-keys -draft %{ %d!cat<space>\"${newer}\"<ret>d }

            ## If the backup file has to be removed, issue the command once
            ## the current buffer has been saved
            ## If the autorestore_purge_restored option has been unset right after the
            ## buffer was restored, do not remove the backup
            hook -group autorestore buffer BufWritePost '${kak_buffile}' %{
                nop %sh{
                    if [ \"\${kak_opt_autorestore_purge_restored}\" = true ];
                    then
                        rm -f \"${buffer_dirname}/.${buffer_basename}.kak.\"* 
                    fi
                }
            }
        "
    
----
```

**Comment by vbauerster** [comment]
Just in case:
```
❯ tmux -V
tmux 2.6
```

**Comment by nochiel** [comment]
Please could you test this:
```diff
diff --git a/rc/extra/tmux-repl.kak b/rc/extra/tmux-repl.kak
index c1b9bcfe..ec9b0883 100644
--- a/rc/extra/tmux-repl.kak
+++ b/rc/extra/tmux-repl.kak
@@ -7,18 +7,20 @@ hook global KakBegin .* %{
             VERSION_TMUX=$(tmux -V)
             VERSION_TMUX=$(expr "${VERSION_TMUX}" : 'tmux \([0-9]*\|master\)')
 
-            if [ "${VERSION_TMUX}" = "master" ] \
-                || [ "${VERSION_TMUX}" -ge 2 ]; then
-                echo "
-                    alias global repl tmux-repl-horizontal
-                    alias global send-text tmux-send-text
-                "
-            else
-                echo "
-                    alias global repl tmux-repl-disabled
-                    alias global send-text tmux-repl-disabled
-                "
-            fi
+            [ -n "${VERSION_TMUX}" ]  && {
+                if [ "x${VERSION_TMUX}" = "xmaster" ] \
+                    || [ "${VERSION_TMUX}" -ge 2 ]; then
+                    echo "
+                        alias global repl tmux-repl-horizontal
+                        alias global send-text tmux-send-text
+                    "
+                else
+                    echo "
+                        alias global repl tmux-repl-disabled
+                        alias global send-text tmux-repl-disabled
+                    "
+                fi
+            }
         fi
     }
 }

```

**Comment by lenormf** [comment]
What is that patch supposed to demonstrate? It looks like the value stored in `$VERSION_TMUX` isn't quite handled by `expr`, I think adding `set +x` at the head of the shell scope would be more helpful.

**Comment by nochiel** [comment]
A quick test to check that the variable actually is actually set after the call to expr.

**Comment by nochiel** [comment]
> adding set +x at the head of the shell scope would be more helpful.

I don't get a trace when using set +x in kak. How do you use it?

**Comment by lenormf** [comment]
I think the problem is that `tmux -V` returns a value that's not handled by the `expr` regex, even though the info given by @vbauerster seems to not point that way.

To use the `x` flag, just append `set +x` after any `%sh{` scope opening token:

```
%sh{ set +x;
    VERSION_TMUX=$(tmux -V)
    VERSION_TMUX=$(expr "${VERSION_TMUX}" : 'tmux \([0-9]*\|master\)')
[…]
```

The shell should print on `stderr` what value was assigned to `$VERSION_TMUX`.

**Comment by mawww** [comment]
What shell/OS are you using ? Here on both bash and dash, the `expr` call seems to do the right thing with "tmux 2.6" as input. It seems it fails to do that for some reason on your system.

**Comment by lenormf** [comment]
I have a feeling that the `tmux` binary that is used in the shell scope isn't the one available in a regular user shell, hence the `set +x` flag.

**Comment by vbauerster** [comment]
```
❯ zsh --version
zsh 5.4.2 (x86_64-apple-darwin17.3.0)
```

**Comment by lenormf** [comment]
I think what you want here is rather:

```bash
if [ "${VERSION_TMUX}" = "master" ] \
   || expr "${VERSION_TMUX}" \>= 2 >/dev/null; then
```

Won't work that well otherwise, I reckon:

```bash
> expr 2.6 \>= 2; echo $?
1
0

> expr 2.6 \>= 3; echo $?
0
1

> [ $(false) ]; echo $?
1

> [ $(true) ]; echo $?
1
```

**Comment by mawww** [comment]
I'd use cut instead of awk for a simple case like that. Then use shell expansion to get only the major version

So, something like `VERSION_TMUX=$(tmux -V | cut -d' ' -f2)` then $(expr ${VERSION_TMUX%%.*} \>= 2)`

**Comment by mawww** [comment]
I believe this should be fixed by bb124895b73357e7834e0bebbebe0678946c5bb4
